### PR TITLE
[#7443] Make `copy_data_object()` honor `copy_options::none` correctly (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Not yet tested.

### CODE REVIEW

File: unit_tests/src/filesystem/test_filesystem.cpp [hunk 1]
```diff
@@ -769,4 +770,62 @@ TEST_CASE("filesystem")
...
+            default_transport tp{conn};
+            odstream{tp, data_object} << contents;
+        }
+
+        REQUIRE(fs::client::is_data_object(conn, data_object));
+
+        // Create a copy of the data object using copy_options::none.
+        // The assertion that follows specifically proves #7443 is resolved.
+        const auto new_data_object = sandbox / "data_object.txt.copy";
+        REQUIRE(fs::client::copy_data_object(conn, data_object, new_data_object, fs::copy_options::none));
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply unit_tests/src/filesystem/test_filesystem.cpp:790]: Sure.

[the remark this answers is not in the document]
Comment (korydraughn) [reply unit_tests/src/filesystem/test_filesystem.cpp:790]: Done.

**Comment by korydraughn** [comment]
Will test soon.

**Comment by korydraughn** [comment]
All unit tests pass.

Running core tests.

**Comment by korydraughn** [comment]
All core tests passed.

**Review by alanking — APPROVED** [review]
I think we are good to go here

**Comment by korydraughn** [comment]
#'d

**Comment by korydraughn** [comment]
Hold please

**Comment by korydraughn** [comment]
Ok. It's good to go. I wanted to see clang-format turn green.

**Pull request merged**